shared/js/loadApp.js

```diff
@@ -12,7 +15,65 @@ function loadStyle(name) {
 }
 
 module.exports = function (callback) {
-  if (window.dashboard && dashboard.project) {
+  if (!appOptions.channel && appOptions.publicCaching) {
+    var lastAttemptLoaded = false;
+
+    var loadLastAttemptFromSessionStorage = function () {
+      if (!lastAttemptLoaded) {
+        lastAttemptLoaded = true;
+
+        // Load the locally-cached last attempt (if one exists)
+        setLastAttemptUnlessJigsaw(dashboard.clientState.sourceForLevel(
+            appOptions.scriptName, appOptions.serverLevelId));
+
+        callback();
+      }
+    };
+
+    $.ajax('/api/user_progress/' + appOptions.scriptName + '/' + appOptions.stagePosition + '/' + appOptions.levelPosition).done(function (data) {
+      // Merge progress from server (loaded via AJAX)
+      var serverProgress = data.progress;
```

Comment:
After HOC we'll want to refactor this into a helper function we can more easily test.